src/util/ActionProgramUtils.ts

```diff
@@ -261,66 +266,97 @@ export const makeAaveCloseAction = async ({
-  } else {
-    // If the user does not have enough debt balance to fully cover the principal,
-    // first repay with the maximum debt balance such that the following repay
-    // with collateral will not fail due to not meeting paraswap minimums
-    repayWithCollateralNativeAmount = max(sub(debt.nativeAmount, debtBalanceNativeAmount), MINIMUM_PARASWAP_AMOUNT)
-    repayWithBalanceNativeAmount = sub(debt.nativeAmount, repayWithCollateralNativeAmount)
-  }
 
-  // Construct repay with balance
-  if (!zeroString(repayWithBalanceNativeAmount))
-    seqAction.actions.push({
-      type: 'loan-repay',
-      nativeAmount: repayWithBalanceNativeAmount,
-      borrowPluginId,
-      tokenId: debtTokenId,
-      walletId: wallet.id
-    })
-
-  // Construct repay with collateral
-  if (!zeroString(repayWithCollateralNativeAmount))
-    seqAction.actions.push({
-      type: 'loan-repay',
-      nativeAmount: repayWithCollateralNativeAmount,
-      borrowPluginId,
-      fromTokenId: collateralTokenId,
-      tokenId: debtTokenId,
-      walletId: wallet.id
-    })
+  if (debtTokenId != null) {
+    const { currencyCode: collateralCurrencyCode, denominations: collateralDenoms } =
+      collateralTokenId == null ? currencyInfo : allTokens[collateralTokenId] ?? {}
+    const { currencyCode: debtCurrencyCode, denominations: debtDenoms } = allTokens[debtTokenId] ?? {}
+    const collateralDenom = collateralDenoms.find(denom => denom.name === collateralCurrencyCode)
+    const collateralMultiplier = collateralDenom?.multiplier ?? '1'
+    const collateralFiat = convertCurrencyFromExchangeRates(
+      exchangeRates,
+      collateralCurrencyCode,
+      isoFiatCurrencyCode,
+      convertNativeToExchange(collateralMultiplier)(collateral.nativeAmount)
+    )
+    const debtDenom = debtDenoms.find(denom => denom.name === debtCurrencyCode)
+    const debtMultiplier = debtDenom?.multiplier ?? '1'
+    const principalFiat = convertCurrencyFromExchangeRates(
+      exchangeRates,
+      debtCurrencyCode,
+      isoFiatCurrencyCode,
+      convertNativeToExchange(debtMultiplier)(debt.nativeAmount)
+    )
+    const debtBalanceNativeAmount = wallet.balances[debtCurrencyCode]
+    const debtBalanceFiat = convertCurrencyFromExchangeRates(
+      exchangeRates,
+      debtCurrencyCode,
+      isoFiatCurrencyCode,
+      convertNativeToExchange(debtMultiplier)(debtBalanceNativeAmount)
+    )
+
+    // First, calculate the remaining principal after a payment is made from the
+    // debt asset balance
+    const postRepayPrincipalFiat = sub(principalFiat, debtBalanceFiat)
+
+    // Find maximum amount of collateral that can be swapped while remaining
+    // above the liquidation threshold (74%), where:
+    // Pr: Principal remaining after initial payment above
+    // C: Collateral
+    // Cs: Collateral to swap (maxSwappableCollateral)
+    //
+    // Pr / (C - Cs) = 0.74
+    // Solved for Cs:
+    // Cs = -((50Pr - 37C) / 37)
```

Comment:
I learnt in my limited exposure to formal maths that the ideal form is a polynomial:

`Cs = -1/.74 * Pr + C`

This is nice because now you only have one constant to deal with.

Full solve:
```
Pr / (C -Cs) = 0.74
1 / (C-Cs) = 0.74/Pr
C -Cs = Pr / .74
-Cs = Pr / .74 - C
Cs = -(Pr / .74 - C)
Cs = -Pr / .74 + C
Cs = Pr * -1/.74 + C
Cs = -1/.74 * Pr + C
```